packages/allow-scripts/src/index.js

```diff
@@ -452,6 +452,17 @@ async function loadAllPackageConfigurations({ rootDir }) {
       (name) => Object.prototype.hasOwnProperty.call(depScripts, name)
     )
 
+    if (
+      !lifeCycleScripts.includes('preinstall') &&
+      !lifeCycleScripts.includes('install')
+    ) {
+      try {
+        await fs.stat(path.join(filePath, 'binding.gyp'))
+        lifeCycleScripts.unshift('install')
+        depScripts.install = 'node-gyp rebuild'
+      } catch {}
+    }
+
```

Comment:
This should be the same but simpler?
```suggestion
    if (
      !lifeCycleScripts.includes('preinstall') &&
      !lifeCycleScripts.includes('install') &&
      existsSync(path.join(filePath, 'binding.gyp'))
    ) {
        lifeCycleScripts.unshift('install')
        depScripts.install = 'node-gyp rebuild'
    }

```

Will also need this up top:

```
const { promises: fs, existsSync } = require('node:fs')
```